mattermod.go

```diff
@@ -23,21 +22,28 @@ func main() {
 
 	config, err := server.GetConfig(configFile)
 	if err != nil {
-		err = errors.Wrap(err, "unable to load server config")
 		mlog.Error("unable to load server config", mlog.Err(err))
-		panic(err)
+		os.Exit(1)
 	}
 	server.SetupLogging(config)
 
 	mlog.Info("Loaded config", mlog.String("filename", configFile))
+	s := server.New(config)
 
-	s, err := server.New(config)
-	if err != nil {
-		panic("failed creating server")
-	}
-
-	s.Start()
-	defer s.Stop()
+	mlog.Info("Starting Mattermod Server")
+	go func() {
```

Comment:
0/5, but it felt cleaner to spawn the goroutine internally from inside `s.Start`. I know there are guidelines that say making calls async should be left to caller. But in general, starting a server is always async from what I have seen.

Reply:
Actually I just didn't wanted to `os.Exit()` in the `server.go` :)

Reply:
I don't think you need to. The process is already on the way to its exit. We just need to log it. That's all. Unless I have missed something.

Reply:
Let's say if the port is used by some other process, the app will hang in there and will need to be interrupted.

Reply:
That's a good point. Personally, I am fine with os.Exit in server.go. But 0/5, up to you. This is already good to go from my side.

Reply:
I see, let's go with your suggestion. I will re-consider this in the `main()` overhaul 👍